homer/datadog_checks/homer/check.py

```diff
@@ -0,0 +1,112 @@
+import psutil
+import psycopg2
+
+from datadog_checks.base import AgentCheck
+
+
+class ConnNoop(object):
+    def close(self):
+        pass
+
+
+class HomerCheck(AgentCheck):
+    # this will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'homer'
+
+    def __init__(self, name, init_config, instances):
+        super(HomerCheck, self).__init__(name, init_config, instances)
+
+    def check(self, instance):
+        """
+        Run the homer service checks and send metrics to datadog API
+
+        :param instance:    The current instance configuration
+        :type instance:     object|dict
+        :return:            None
+        :rtype:             None
+        """
+
+        # validate instance configs
+        homer_service_checks = instance.get('homer_service_checks')
+        if homer_service_checks is None:
+            self.service_check(
+                "services_up", AgentCheck.WARNING, message='agent misconfiguration: homer_service_checks is not valid'
+            )
+            return None
+        homer_config_db = instance.get("homer_config_db")
+        if homer_config_db is None:
+            self.service_check(
+                "metrics_up", AgentCheck.WARNING, message='agent misconfiguration: homer_config_db is not valid'
+            )
+            return None
+        homer_data_db = instance.get("homer_data_db")
+        if homer_data_db is None:
+            self.service_check(
+                "metrics_up", AgentCheck.WARNING, message='agent misconfiguration: homer_data_db is not valid'
+            )
+            return None
+
+        # check if homer processes are alive
+        down_services = []
+        for service in homer_service_checks:
+            if not HomerCheck.isProcAlive(service):
+                down_services.append(service)
+        if len(down_services) != 0:
+            self.service_check(
+                "services_up", AgentCheck.CRITICAL, message=f'found homer services that are down: {down_services}'
+            )
+            return None
```

Comment:
```suggestion
            return
```

Also you could log the message using `self.log.error`